### PR TITLE
Remove calls to `TestCase::iniSet()` and calls to deprecated methods of `MockBuilder`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Last round (before new deprecations 🙂). All deprecations [listed here](https://github.com/sebastianbergmann/phpunit/blob/main/DEPRECATIONS.md) should be gone.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Tests/Extension/HttpKernelExtensionTest.php [hunk 1]
```diff
@@ -87,11 +87,17 @@ public function testGenerateFragmentUri()
         $this->assertSame('/_fragment?_hash=PP8%2FeEbn1pr27I9wmag%2FM6jYGVwUZ0l2h0vhh2OJ6CI%3D&amp;_path=template%3Dfoo.html.twig%26_format%3Dhtml%26_locale%3Den%26_controller%3DSymfonyBundleFrameworkBundleControllerTemplateController%253A%253AtemplateAction', $twig->render('index'));
     }
 
-    protected function getFragmentHandler($return)
+    protected function getFragmentHandler($returnOrException)
```
Comment (OskarStark) [inline src/Symfony/Bridge/Twig/Tests/Extension/HttpKernelExtensionTest.php]: add return type?

Comment (alexandre-daubois) [reply src/Symfony/Bridge/Twig/Tests/Extension/HttpKernelExtensionTest.php]: Why not 🙂

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php [hunk 2]
```diff
@@ -352,11 +362,13 @@ public function testReverseTransformWrapsIntlErrorsWithExceptions()
...
         }
 
-        $this->iniSet('intl.use_exceptions', 1);
+        $initialUseExceptions = ini_set('intl.use_exceptions', 1);
 
         $this->expectException(TransformationFailedException::class);
         $transformer = new DateTimeToLocalizedStringTransformer();
         $transformer->reverseTransform('12345');
+
+        ini_set('intl.use_exceptions', $initialUseExceptions);
```
Comment (derrabus) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php]: Make sure that the ini value is reset even if the test fails. You can achieve that by wrapping the code into a `try`/`finally` block. You can also consider moving the cleanup part to the `tearDown()` method.

```suggestion
        $initialUseExceptions = ini_set('intl.use_exceptions', 1);

        $this->expectException(TransformationFailedException::class);
        
        try {
            $transformer = new DateTimeToLocalizedStringTransformer();
            $transformer->reverseTransform('12345');
        } finally {
            ini_set('intl.use_exceptions', $initialUseExceptions);
        }
```

File: src/Symfony/Component/Console/Tests/Fixtures/MockableAppliationWithTerminalWidth.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Tests\Fixtures;
+
+use Symfony\Component\Console\Application;
+
+class MockableAppliationWithTerminalWidth extends Application
+{
+    public function getTerminalWidth()
```
Comment (derrabus) [inline src/Symfony/Component/Console/Tests/Fixtures/MockableAppliationWithTerminalWidth.php]: Return type?

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Console/Tests/Fixtures/MockableAppliationWithTerminalWidth.php (hunk 3 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Console/Tests/Fixtures/MockableAppliationWithTerminalWidth.php]: Added where missing and relevant

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php]: You're right. I wrapped all calls to `ini_set()` with `try/finally`

File: src/Symfony/Component/Routing/Tests/Loader/ObjectLoaderTest.php [hunk 4]
```diff
@@ -109,6 +108,13 @@ protected function getObject(string $id): object
     }
 }
 
+class stdClassWithLoadRoutesMethod extends \stdClass
+{
+    public function loadRoutes()
+    {
+    }
+}
```
Comment (derrabus) [inline src/Symfony/Component/Routing/Tests/Loader/ObjectLoaderTest.php]: This does not have to be a `stdClass`, does it.

```suggestion
interface CustomRouteLoader
{
    public function loadRoutes();
}
```

Comment (alexandre-daubois) [reply src/Symfony/Component/Routing/Tests/Loader/ObjectLoaderTest.php]: Updated, it is far more elegant 👍

File: src/Symfony/Component/Security/Guard/Tests/GuardAuthenticatorHandlerTest.php [hunk 5]
```diff
@@ -159,12 +159,14 @@ public function testSessionStrategyIsNotCalledWhenStateless()
...
-        $sessionFactory->expects($this->never())
-            ->method('__invoke');
+        $this->expectNotToPerformAssertions();
+
+        $sessionFactory = new class() extends \stdClass {
+            public function __invoke()
+            {
+                throw new \LogicException('This should not be called');
+            }
+        };
```
Comment (derrabus) [inline src/Symfony/Component/Security/Guard/Tests/GuardAuthenticatorHandlerTest.php]:
```suggestion
        $sessionFactory = static function (): void {
            throw new \LogicException('This should not be called');
        };
```

File: src/Symfony/Component/Security/Http/Tests/Firewall/ContextListenerTest.php [hunk 6]
```diff
@@ -366,8 +366,14 @@ public function testWithPreviousNotStartedSession()
...
-        $usageReporter = $this->getMockBuilder(\stdClass::class)->addMethods(['__invoke'])->getMock();
-        $usageReporter->expects($this->never())->method('__invoke');
+        $this->expectNotToPerformAssertions();
+
+        $usageReporter = new class() {
+            public function __invoke()
+            {
+                throw new \LogicException('This should not be called');
+            }
+        };
```
Comment (derrabus) [inline src/Symfony/Component/Security/Http/Tests/Firewall/ContextListenerTest.php]:
```suggestion
        $usageReporter = static function (): void {
            throw new \LogicException('This should not be called');
        };
```

**Review by derrabus — COMMENTED** [review]
Sorry for reviewing in small chunks. This is hopefully the last batch of nit-picks. 😓

**Comment by alexandre-daubois** [comment]
No worries, these comments bring elegant code so that's nice to update 😄

**Review by derrabus — APPROVED** [review]
fabbot's comment about the closure syntax is valid, that was my bad. 😓 

Apart from that: LGTM.

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you @alexandre-daubois.

**Pull request merged**